robottelo/ui/hosts.py

```diff
@@ -12,82 +12,98 @@ def _configure_hosts(self, arch=None, custom_ptable=None, cv=None,
                          domain=None, env=None, host_group=None, ip_addr=None,
                          lifecycle_env=None, mac=None, media=None, os=None,
                          ptable=None, puppet_ca=None, puppet_master=None,
+                         puppet_module=None, reset_puppetenv=True,
                          resource=None, root_pwd=None, subnet=None):
+        strategy1, value1 = locators['host.select_puppetmodule']
+        strategy2, value2 = locators['host.select_puppetclass']
         # Host tab
-        if lifecycle_env is not None:
+        if lifecycle_env:
```

Comment:
Empty strings evaluate to false. This change prevents us from passing empty strings to the UI, which could be quite useful for testing.